src/_pytest/compat.py

```diff
@@ -76,18 +77,41 @@ def is_async_function(func: object) -> bool:
     return iscoroutinefunction(func) or inspect.isasyncgenfunction(func)
 
 
-def getlocation(function, curdir: Optional[str] = None) -> str:
+class CodeLocation(NamedTuple):
+    path: Path
+    lineno: int
+
+
+# TODO: integrate after pytest 3.6.0 has been dropped
+def CodeLocation__str__(self: CodeLocation) -> str:
+    """Python 3.6.0 hack for NamedTuple __str__"""
+    return f"{self.path}:{self.lineno}"
+
+
+setattr(CodeLocation, "__str__", CodeLocation__str__)
+
+
+def getlocation(
+    function, *, relative_to: Optional[Path], allow_escape: bool
```

Comment:
A docstring would be helpful, it was not very clear to me from the name what `allow_escape` means.